### PR TITLE
Sidebare preview fixes

### PR DESCRIPTION
**Pull request by icewind1991**
- square images as no longer handled as landscape
- height for landscape image previews now based on the sidebar width instead of constant (16/9 atm, not sure if that's optimal)
- Fix invalid url for previews for images in subfolders
- portrait previews have their regular margin since we dont fit them anyways
- landscape previews are cropped to the center instead of the top

cc @jancborchardt @oparoz

### CODE REVIEW

**Comment by icewind1991** [comment]
Also, currently any images which is smaller than 75x75 (the size of the small preview for non-images) the preview is shown as the small images.

We might want to increase this limit to something like 100-150px to prevent small previews looking out of place

**Comment by oparoz** [comment]
eps

* Why is it a small thumbnail when the image is a large one?
* There needs to be more space between the image name and the details on the line below it

**Comment by oparoz** [comment]
jpg

* Too much space below the image

**Comment by oparoz** [comment]
png

* Small png still blown up

**Comment by oparoz** [comment]
white

* White images still invisible

**Comment by oparoz** [comment]
Some problems could come from Preview, others from the lack of an automatic, proper background selection for previews (also affects Gallery)

**Comment by icewind1991** [comment]
> Why is it a small thumbnail when the image is a large one?

eps not recognized as image by mimetype, will fix

> Small png still blown up

That should have been fixed? are you sure you refreshed? can you link me the faulty image

**Comment by icewind1991** [comment]
Not going to try and fix the white-on-white problem with this PR since it's a much wider problem with preview

**Comment by oparoz** [comment]
> That should have been fixed? are you sure you refreshed? can you link me the faulty image

Stupid me has fetched new branches, but didn't switch... Will check again. Apologies.

**Comment by icewind1991** [comment]
eps, psd and ia (illustrator) should now properly show as images

**Comment by oparoz** [comment]
png

* Small images are fixed

**Comment by karlitschek** [comment]
nice :+1: :-)

File: apps/files/js/fileinfomodel.js [hunk 1]
```diff
@@ -57,7 +57,13 @@
 		 * @return {boolean} true if this is an image, false otherwise
 		 */
 		isImage: function() {
-			return this.has('mimetype') ? this.get('mimetype').substr(0, 6) === 'image/' : false;
+			if (!this.has('mimetype')) {
+				return false;
+			}
+			return this.get('mimetype').substr(0, 6) === 'image/'
```
Comment (oparoz) [inline apps/files/js/fileinfomodel.js]: We need a better approach as the user can install any preview provider he wants and get large previews of his files.

Nothing exists in `core` afaik. In Gallery, there is a harcoded array of supported media types...

**Comment by oparoz** [comment]
So, 3 problems left:
* Spacing below the title
* White images. Should be addressed in separate PR
* Visual media detection so that they don't get a thumbnail. Could be addressed in another PR which would fix the detection mechanism put in place here

**Comment by icewind1991** [comment]
> Spacing below the title

Out of scope of this PR since it's unrelated to the large previews

**Comment by oparoz** [comment]
OK

File: apps/files/js/fileinfomodel.js (hunk 1 above)
Comment (nickvergessen) [inline apps/files/js/fileinfomodel.js]: I only know https://github.com/owncloud/core/blob/e7da416da09b2492174668254ce2449b5afc1682/lib/private/previewmanager.php#L157

Comment (oparoz) [inline apps/files/js/fileinfomodel.js]: Actually, why isn't the sidebar using the same mechanism used for thumbnails? There are previews for everything and the fallback is the media type icon. I think it works by using `forceicon=0`

Comment (oparoz) [inline apps/files/js/fileinfomodel.js]: Why not get a large office/pdf/font preview. That's something a lot of people have asked for. They don't want to use the slideshow and there is no grid view, so they will rely on this to get a better view of what the file is about.

**Comment by icewind1991** [comment]
rebased

**Comment by icewind1991** [comment]
@oparoz @jancborchardt can this be merged?

**Comment by jancborchardt** [comment]
Filetype icons in the sidebar are suddenly smaller.

File: apps/files/js/mainfileinfodetailview.js [hunk 2]
```diff
@@ -179,6 +137,66 @@
...
+
+					// only set background when we have an actual preview
+					// when we dont have a preview we show the mime icon in the error handler
+					$iconDiv.css({
+						'background-image': 'url("' + previewUrl + '")',
+						'height': targetHeight
+					});
+				}.bind(this),
+				error: function () {
+					$iconDiv.removeClass('icon-loading');
```
Comment (georgehrke) [inline apps/files/js/mainfileinfodetailview.js]: missing icon-32?

Comment (icewind1991) [inline apps/files/js/mainfileinfodetailview.js]: Good catch, fixed

**Comment by icewind1991** [comment]
> Filetype icons in the sidebar are suddenly smaller.

Fixed

**Comment by oparoz** [comment]
There is still the problem of the visual media detection mechanism.

**Comment by icewind1991** [comment]
> There is still the problem of the visual media detection mechanism.

Seperate PR, we need to get these changes in for 8.2

**Comment by jancborchardt** [comment]
Looks good to me now, I also fixed https://github.com/owncloud/core/issues/18981

File: apps/files/js/filelist.js [hunk 3]
```diff
@@ -327,7 +327,7 @@
 			// and contain existing models that can be used.
 			// This method would in the future simply retrieve the matching model from the collection.
 			var model = new OCA.Files.FileInfoModel(this.elementToFile($tr));
-			if (!model.has('path')) {
+			if (!model.get('path')) {
```
Comment (PVince81) [inline apps/files/js/filelist.js]: Empty string is a valid path (the root). But usually it should be "/".
Did you find any case where this happens ?

**Comment by PVince81** [comment]
Looks fine :+1: 

@icewind1991 @oparoz please raise separate tickets/PR for the other issues that were mentionned to be fixed separately

**Comment by oparoz** [comment]
Last PR regarding issues encountered in this one: https://github.com/owncloud/core/issues/19054

**Comment by jancborchardt** [comment]
Rebase and merge @icewind1991?

**Comment by PVince81** [comment]
No need to rebase.
The grey icon is only because CI hasn't run yet...

(got confused by this too)

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**